app/api/route/user_route.py

```diff
@@ -27,8 +28,9 @@ def user_post(user_params: users.CreateUserParams) -> dict:
     """
     logger.info("POST /v1/user")
 
-    with api_context_manager() as api_context:
-        user = user_service.create_user(user_params, api_context)
+    db = api.db.get_db(current_app)
+    with db.get_session() as db_session:
```

Comment:
Do you think it would be useful to have a shortcut method that lets you just do `with api.db_session()...` so you don't need to unpack as much?

Reply:
It would be more convenient for the developer / save one line of code, but one feature I like about the current approach is that it makes it explicit that this file depends on the `api.db` module. If we switched it to `api.db_session` then it's not clear from looking at the imports that this module depends on functionality in `api.db`. I was thinking that longer term it could be beneficial to add a linter/test/check that helps enforce architectural patterns, such as modules in `services` should not import modules in `routes`, etc. so having dependencies by explicit is a nice bonus IMO. What do you think?

Reply:
What about importing as `import api.db as db`, I actually at first assumed that db was some object in the api module, when the api isn't actually relevant here.

Could also shorten this as 
```
with db.get_db(current_app).get_session() as db_session:
```
Which still feels very verbose.

Reply:
> I actually at first assumed that db was some object in the api module, when the api isn't actually relevant here.

that's a good point, we don't often import nested modules like this, so i can see the confusion.

i changed it to `import api.db as db` like you suggested, but then i also adopted the convention that instances of `DBClient` should be named `db_client` rather than `db`, to avoid confusion between the `db` client and the `db` module.

i also shortened the line as you suggested

lemme know your thoughts